erpnext/selling/doctype/sales_order/sales_order.js

```diff
@@ -226,10 +226,43 @@ erpnext.selling.SalesOrderController = erpnext.selling.SellingController.extend(
 						fields: fields,
 						primary_action: function() {
 							var data = d.get_values();
+
+							data_object_items = [];
+							for(i = 0; i < data["items"].length; i++) {
+								data_object_items.push(data.items[i].sales_order_item);
+							}
+
+							visible_rows = $(".modal-dialog:has(div[data-fieldtype=\"Table\"][data-fieldname=\"items\"]) .grid-body .grid-row");
+
+							visible_items = [];
+
+							$(visible_rows).each(function() {
+								visible_items.push($(this).find("div[data-fieldname=\"sales_order_item\"] .static-area.ellipsis").text());
```

Comment:
Same comment as above check the dialog, field_group and friends for cleaner ways of writing selectors.